### DISCUSSION TITLE
`model_validate` vs `**data`

### DISCUSSION THREAD

**Discussion by jtfidje** [pydantic/pydantic#9676] [Question] [answered]
Is there a performance difference between using `SomeModel.model_validate` over `SomeModel(**data)` ? 🙂

**Comment by gbaian10** [comment]
I have the exact same question.
Which approach is recommended?

**Comment by Viicos** [comment, marked as answer]
It should be identical (they both call the same validator method). `model_validate` gives you more options though (extra context, etc).

**Reply by jtfidje** [reply]
Thanks :) Even though they might be identical - do you prefer one over the other, and if so why?

**Reply by vadim-su** [reply]
@jtfidje Explicit is better than implicit. Also, `model.model_validate(some_raw_data)` and `model_inst.model_dump(mode='json')` look more consistent than `model(**some_raw_data)` in my opinion.

**Reply by vadim-su** [reply]
I use declaration like
```py
model_inst = Model(
    field_a='a',
    field_b=1,
)
```
only when I want to set fields directly (from a function args for instance).

**Reply by Viicos** [reply]
Yep, usually I also use direct instantiation when I manually specify each argument as well. Otherwise I tend to use `model_validate` if I have a full dict.

**Comment by MarcBresson** [comment]
Hello, here is a benchmark for this:

```python
from pydantic import BaseModel
import timeit

class A(BaseModel):
    a: int
    b: int
    c: float
    d: str
    e: list[int]

data = {
    "a": 8.0,
    "b": "3",
    "c": 0.44424235237894567890,
    "d": "coucou",
    "e": [1, 2, 3, 1278, 2.0, 3.0, 4.0]
}

def parse_with_validate():
    return A.model_validate(data)

def parse_with_dict_decomposition():
    return A(**data)

results = []

functions_to_test = (parse_with_validate, parse_with_dict_decomposition)

for f in functions_to_test:
    timings = timeit.repeat(lambda: f(), repeat=10, number=100000)

    timings = np.array(timings)

    results.append((timings.mean() / 100000, timings.std() / 100000))

results

#  average speed (second) std (second)
# (1.022037998016458e-06, 7.42553531902382e-08),
# (9.361435420578346e-07, 2.995572902708461e-08)
```
